x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java

```diff
@@ -536,16 +537,15 @@ protected QueryTranslation asQuery(BinaryComparison bc, boolean onAggs) {
                 //
                 // Agg context means HAVING -> PipelineAggs
                 //
-                ScriptTemplate script = bc.asScript();
```

Comment:
Any reason why it got moved down? Did it fail when used against a `FieldAttribute` (which suggest a bug in the pipeline/querytranslator)?

Reply:
There was no bug there, I just thought that it's better to called the `asScript()` when actually needed and not beforehand. If you have any objection I can revert.